objects/src/assets/mod.rs

```diff
@@ -22,21 +24,22 @@ use core::{fmt, ops::Deref};
 /// The methodology for constructing fungible and non-fungible assets is described below.
 ///
 /// # Fungible assets
-/// The first 3 elements of a fungible asset are set to the ID of the faucet which issued the
-/// asset. This guarantees the properties described above (the first bit is ONE, and the 32 least
-/// significant bits of the 3rd element are all ZEROs).
+/// The most significant element of a fungible asset is set to the ID of the faucet which issued
+/// the asset. This guarantees the properties described above (the first bit is ONE).
 ///
-/// The last element is set to the amount of the asset. This amount cannot be greater than
-/// 2^63 - 1 and thus requires 63-bits to store.
+/// The least significant element is set to the amount of the asset. This amount cannot be greater
+/// than 2^63 - 1 and thus requires 63-bits to store.
+///
+/// Element 1 and 2 are set to ZERO.
 ///
 /// # Non-fungible assets
 /// The 4 elements of non-fungible assets are computed as follows:
 /// - First the asset data is hashed. This compresses an asset of an arbitrary length to 4 field
-///   elements.
-/// - The result of step 1 is hashed together with the ID of the faucet which issued the asset.
+///   elements: [d0, d1, d2, d3].
+/// - d1 is then replaced with the faucet_id which issues the asset: [d0, faucet_id, d2, d3].
 ///   This guarantees that finding two assets with the same hash is infeasible.
```

Comment:
We should probably change the second sentence because it is not immediately obvious how/why it follows from the first.

Reply:
done